src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -233,78 +204,60 @@ def _format_resource_record_changes(self, action, resource_record):
             change_id (str): Change ID of the Changes resource.
-            base_changes_url (str): The URL prefix to use to check on changes.
-
         Returns:
             True if the changes are done, False if anything else.
         """
-        change_id_url = f'{base_changes_url}/{change_id}'
         timeout = datetime.timedelta(seconds=self.publish_wait_timeout)
         start = datetime.datetime.now()
         end = start + timeout
 
         while datetime.datetime.now() < end:
-            resp = await self.http_client.get_json(change_id_url)
-            if resp['status'] == GDNSPublisher.DNS_CHANGES_DONE:
+            if await self.dns_client.is_change_done(zone):
                 return True
             await asyncio.sleep(1)
 
         return False
 
-    async def _dispatch_changes(self, resource_record, managed_zone, action,
+    async def _dispatch_changes(self, resource_record, zone, action,
                                 logger):
         """Publish changes for one record, making sure they are completed.
 
         Args:
             resource_record (dict): A resource record to change.
-            managed_zone (str): The managed zone name in Google DNS.
+            zone (str): The DNS zone name.
             action (str): The action to take (e.g. 'additions').
             logger (object): An object to log with.
 
         Raises:
             GCPPublishRecordTimeoutError if publishing of records exceeds wait
                 timeout.
         """
-        base_changes_url = GDNSPublisher.BASE_CHANGES_ENDPOINT.format(
-            version=self.api_version, project=self.project,
-            managedZone=managed_zone)
-        resource_records_url = GDNSPublisher.RESOURCE_RECORDS_ENDPOINT.format(
-            project=self.project, managedZone=managed_zone)
-
-        changes_to_publish = self._format_resource_record_changes(
+        changes = self._format_resource_record_changes(
             action, resource_record)
         try:
-            change_id = await self._publish_changes(
-                changes_to_publish, base_changes_url)
-
+            change_id = await self.dns_client.publish_changes(zone, changes)
         except exceptions.GCPHTTPResponseError as e:
             if e.status != 409:
                 raise e
 
             msg = ('Conflict found when publishing records. Handling and '
                    'retrying.')
             logger.info(msg)
-            changes_to_publish = await self._handle_additions_conflict(
-                changes_to_publish, resource_records_url)
-            change_id = await self._publish_changes(changes_to_publish,
-                                                    base_changes_url)
-
-        change_complete = await self._changes_published(change_id,
-                                                        base_changes_url)
-        if not change_complete:
+
+            deletions = await self._get_conflicting_rrsets(zone, changes)
```

Comment:
is this accurate that it's only deletions? https://github.com/spotify/gordon-gcp/pull/73/commits/84dea31582fb68488230f2c7d2eb5c03880faa30#diff-8f53676280c6a88b173ebd83ba0803c0R199 says dict containing original additions plus deletions

Reply:
I changed the API of the method a bit. Instead of passing in a changes dict, and adding a deletions to it inside the function, we get just the deletions back here, and add them to the changes dict on the line below. 

It made it easier for me to understand what was going on, but it's not super important to me, if it makes it more confusing for others.

Reply:
That docstring is wrong now.